File: BennuLib/IO/IDecoder.vb
Imports System.IO
Imports System.Collections.Generic


Public Interface IDecoder(Of T)
    ReadOnly Property SupportedExtensions As IEnumerable(Of String)
    Function TryDecode(input As Stream, ByRef decoded As T) As Boolean
    Function Decode(input As Stream) As T
End Interface

File: BennuLib.Sharp/IO/IDecoder.cs
using System.Collections.Generic;
using System.IO;

namespace BennuLib
{


	public interface IDecoder<T>
	{
		IEnumerable<string> SupportedExtensions { get; }
		bool TryDecode(Stream input, out T decoded);
		T Decode(Stream input);
	}
}

